[Lee Jae Ho] iP

jhlee1997 opened the pull request:
### DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jhlee1997 committed:
Level 1. Greet, Echo, Exit

jhlee1997 committed:
Added Duke.java

jhlee1997 committed:
Added Task.java and Duke.java

jhlee1997 committed:
Modified Duke.java, Task.java and Added Deadline.java, Event.java, Todo.java

jhlee1997 committed:
Resolved whitespace issues with runtest.sh

jhlee1997 committed:
Added error control

jhlee1997 committed:
Added delete feature

jhlee1997 committed:
Updated Duke.java with save feature

jhlee1997 committed:
Updated Duke.java, Deadline.java, Event.java to understand dates and times

jhlee1997 committed:
Merge branch 'branch-Level-7'

jhlee1997 committed:
Resolved merge conflicts in Duke.java

jhlee1997 committed:
Created Ui.java, Storage.java. Parser.java, and TaskList.java

jhlee1997 committed:
Created a package Duke

jhlee1997 committed:
Created JUnit Tests for Parser.java and Task.java

jhlee1997 committed:
Created JavaDoc for Duke

jhlee1997 committed:
Follow the CodingStandard

jhlee1997 committed:
Added a Find feature

jhlee1997 committed:
Merge branch 'branch-A-JavaDoc'

jhlee1997 committed:
Resolved merge conflicts with .gitignore

jhlee1997 committed:
Merge branch 'branch-Level-9'

jhlee1997 committed:
Improved coding quality

jhlee1997 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

jhlee1997 committed:
Set up gradle

jhlee1997 committed:
Merge branch 'branch-A-Gradle'

jhlee1997 committed:
created a GUI for Duke

jhlee1997 committed:
Merge branch 'branch-Level-10'

jhlee1997 committed:
Create duke.jar and fix build.gradle

build.gradle file has some wrong settings (e.g. mainClassName
= "seedu.duke.Duke"

These cause duke.jar file generated by Gradle to be faulty.

Shadow version is changed to '5.1.0' under plugins.
mainClassname is changed to "Duke" under application.

This ensures that Gradle is first able to generate duke.jar
file, and second operate with correct functionalities.

Refer to this discussion on dealing with gradlew issues
https://www.py4u.net/discuss/602784

loose-bus-change commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
```
Good use of correct import statement usage

loose-bus-change commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * This class extends Task.
```
You could consider adding a sentence describing what exactly the Deadline class is (maybe what makes it different from other tasks).

loose-bus-change commented on src/main/java/duke/EmptyDescriptionError.java:
```diff
@@ -0,0 +1,16 @@
+package duke;
+/**
+ * This class extends InvalidCommandException. It handles cases when users do not
+ * specify activity description of the task they wish to add.
```
I like this description, very neat and precise!

jhlee1997 committed:
Storage class: add an assert feature

Method .load() lacks an assert feature

This method makes a critical assumption that
every line in the memory text file is formatted
correctly, without being an empty line.

Assert feature ensures that this assumption
stays valid.

This helps in a debugging process, wherein
the text file was not saved properly by
.store() method.

jhlee1997 committed:
Improve Code Quality

loose-bus-change commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    /**
+     * This method will show users that the task has been successfully marked as completed.
```
Perhaps you could consider beginning the statement with a word like 'returns', 'adds', etc. So for this instance, maybe you could say 'shows users that ...'. Other than that, this sentence makes sense perfectly! I like the crisp nature of your descriptions :)

loose-bus-change commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,89 @@
+ * This class manages all requests made to the taskList.
+ */
+public class TaskList {
+    protected ArrayList<Task> tasks;
```
Excellent use of plural form to define a collection!

loose-bus-change reviewed (commented):
LGTM! I like how organized your code is! Good stuff

tsiyuk commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+     */
+    private String handleQuery(String command, String input) {
+        switch (command) {
+            case "bye":
```
Should there be indentation before case clauses? 🤔  The coding standard states that there is no indentation for case clauses.

tsiyuk commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,45 @@
+/**
+ * This class represents each task registered by the user.
+ */
+public class Task {
```
Good OOP! I like that you use tasks as the superclass of todo, deadline, and events to take advantage of polymorphism.

tsiyuk commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @param t The taks being added.
+     */
+    public void add(Task t) {
```
Perhaps using 'task' as the parameter name could add more readability?

tsiyuk commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @return The String form that will be passed to Ui to be shown to the user.
+     */
+    public String list() {
```
Maybe using 'getList' as the method name here?

tsiyuk commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -5,3 +5,98 @@ Hello from
 | |_| | |_| |   <  __/
 |____/ \__,_|_|\_\___|
 
+    ____________________________________________________________
```
I like that you have implemented comprehensive text UI testing!

tsiyuk reviewed (commented):
Good job! The OOP structure is neat! I only have a few suggestions about naming.

jhlee1997 committed:
Merge pull request #1 from jhlee1997/branch-A-Assertions

Storage class: add an assert feature

jhlee1997 committed:
Merge branch 'master' into branch-A-CodeQuality

jhlee1997 committed:
Merge pull request #2 from jhlee1997/branch-A-CodeQuality

Improve Code Quality

jhlee1997 committed:
Add an extension

jhlee1997 committed:
Merge branch 'branch-B-FixedDurationTasks'

jhlee1997 committed:
Add User Guide

jhlee1997 committed:
Updated IP according to peer review

jhlee1997 committed:
Edit Parser.java

jhlee1997 committed:
Update JavaDoc of Parser.java

jhlee1997 committed:
Update JavaDoc
